src/base/dof_map_constraints.C

```diff
@@ -702,7 +702,7 @@ private:
               for (unsigned int qp=0; qp<n_qp; qp++)
                 {
                   // solution at the quadrature point
-                  OutputNumber fineval = 0;
+                  OutputNumber fineval(0);
```

Comment:
Hmm... It's got to do with the Scalar1/Scalar2/Scalar3 stuff I put in recently to handle initialization from mixes of e.g. double literals and long double Reals.  The compiler probably can't infer what Scalar2 and Scalar3 are in the defaults case, so if it can't directly cast from int to T it needs its hand held.